File: python/src/trees/test_invert_binary_tree.py
from invert_binary_tree import invert_tree, invert_tree_recursive
from tree import TreeNode


def test_invert_binary_tree_example_one() -> None:
    root = TreeNode.from_level_order([4, 2, 7, 1, 3, 6, 9])
    assert root is not None
    inverted = invert_tree(root)
    assert inverted is not None
    assert list(inverted.values()) == [4, 7, 2, 9, 6, 3, 1]


def test_invert_binary_tree_example_two() -> None:
    root = TreeNode.from_level_order([2, 1, 3])
    assert root is not None
    inverted = invert_tree(root)
    assert inverted is not None
    assert list(inverted.values()) == [2, 3, 1]


def test_invert_binary_tree_example_three() -> None:
    root = TreeNode.from_level_order([])
    assert root is None
    inverted = invert_tree(root)
    assert inverted is None


def test_invert_binary_tree_recursive_example_one() -> None:
    root = TreeNode.from_level_order([4, 2, 7, 1, 3, 6, 9])
    assert root is not None
    inverted = invert_tree_recursive(root)
    assert inverted is not None
    assert list(inverted.values()) == [4, 7, 2, 9, 6, 3, 1]


def test_invert_binary_tree_recursive_example_two() -> None:
    root = TreeNode.from_level_order([2, 1, 3])
    assert root is not None
    inverted = invert_tree_recursive(root)
    assert inverted is not None
    assert list(inverted.values()) == [2, 3, 1]


def test_invert_binary_tree_recursive_example_three() -> None:
    root = TreeNode.from_level_order([])
    assert root is None
    inverted = invert_tree_recursive(root)
    assert inverted is None

File: typescript/src/trees/test-invert-binary-tree.ts
import { expect, test } from "vitest";
import { TreeNode } from "./tree";
import { invertTree, invertTreeRecursive } from "./invert-binary-tree";

test("invertTree example one", () => {
  const root = TreeNode.from_values([4, 2, 7, 1, 3, 6, 9]);
  const inverted = invertTree(root);
  expect(Array.from(inverted!.values())).toEqual([4, 7, 2, 9, 6, 3, 1]);
});

test("invertTree example two", () => {
  const root = TreeNode.from_values([2, 1, 3]);
  const inverted = invertTree(root);
  expect(Array.from(inverted!.values())).toEqual([2, 3, 1]);
});

test("invertTree example three", () => {
  const root = TreeNode.from_values([]);
  const inverted = invertTree(root);
  expect(inverted).toBeNull();
});

test("invertTreeRecursive example one", () => {
  const root = TreeNode.from_values([4, 2, 7, 1, 3, 6, 9]);
  const inverted = invertTreeRecursive(root);
  expect(Array.from(inverted!.values())).toEqual([4, 7, 2, 9, 6, 3, 1]);
});

test("invertTreeRecursive example two", () => {
  const root = TreeNode.from_values([2, 1, 3]);
  const inverted = invertTreeRecursive(root);
  expect(Array.from(inverted!.values())).toEqual([2, 3, 1]);
});

test("invertTreeRecursive example three", () => {
  const root = TreeNode.from_values([]);
  const inverted = invertTreeRecursive(root);
  expect(inverted).toBeNull();
});
